src/marginPool/marginPool.ts

```diff
@@ -0,0 +1,182 @@
+import { MintInfo, AccountInfo as TokenAccountInfo, NATIVE_MINT, TOKEN_PROGRAM_ID } from "@solana/spl-token"
+import { parseMintAccount, parseTokenAccount } from "../common/accountParser"
+import { PublicKey, SystemProgram, SYSVAR_RENT_PUBKEY } from "@solana/web3.js"
+import { Program } from "@project-serum/anchor"
+import { findDerivedAccount, checkNull } from "../common"
+import { Hooks } from "../common/hooks"
+import { CreatePoolParams, MarginPoolAccountInfo, CreatePoolInfo, MarginPoolConfig } from "./types"
+import { TokenMetadata } from "../marginMetadata"
+
+export interface MarginPoolAddresses {
+  /** The pool's token mint i.e. BTC or SOL mint address*/
+  tokenMint: PublicKey
+  marginPool: PublicKey
+  vault: PublicKey
+  depositNoteMint: PublicKey
+  loanNoteMint: PublicKey
+}
+
+// create marginPool
+// load margin pool,
+// use margin pool,
+// derive accounts
+//decode
+//instructions
+//txns
+export class MarginPool {
+  //FIXME
+  // add obj of addresses for seeds
+  static readonly solTokenMint = NATIVE_MINT
+
+  constructor(
+    public program: Program,
+    public addresses: MarginPoolAddresses,
+    public marginPool: MarginPoolAccountInfo,
+    public vault: TokenAccountInfo,
+    public depositNoteMint: MintInfo,
+    public loanNoteMint: MintInfo,
+    public poolTokenMint: MintInfo,
+    public payer: PublicKey
+  ) {}
+
+  //load multiple pools?
+
+  //add load multiple pools method
+  //change tokenMint to tokenMint
+  /**
+   * Load a Margin Pool Program Account
+   * @param program
+   * @param tokenMint
+   * @returns
+   */
+  static async load(program: Program, tokenMint: PublicKey, payer: PublicKey): Promise<MarginPool> {
```

Comment:
When you see `payer` in the program, you can be safe in setting it to `program.provider.wallet.publicKey`

```suggestion
  static async load(program: Program, tokenMint: PublicKey): Promise<MarginPool> {
```